src/core/include/openvino/core/graph_util.hpp

```diff
@@ -222,14 +222,17 @@ void replace_nodes(const std::shared_ptr<Model>& f,
 /// Topological sort of nodes needed to compute root_nodes
 template <typename T>
 std::vector<std::shared_ptr<Node>> topological_sort(T root_nodes) {
-    std::stack<Node*, std::vector<Node*>> nodes_to_do;
     std::unordered_set<Node*> nodes_done;
     std::unordered_map<Node*, uint8_t /*is_visited*/> nodes_visited;
     std::vector<std::shared_ptr<Node>> result;
 
+    std::vector<Node*> root_nodes_vector;
     for (auto& node : root_nodes) {
-        nodes_to_do.push(node.get());
+        root_nodes_vector.push_back(node.get());
     }
+    std::reverse(root_nodes_vector.begin(), root_nodes_vector.end());
+    std::stack<Node*, std::vector<Node*>> nodes_to_do(root_nodes_vector);
```

Comment:
Additional vector leads to extra copies. In this case `deque` or reverse filling of `stack` were preferable